beacon-chain/blockchain/service.go

```diff
@@ -510,12 +517,38 @@ func (s *Service) updateCustodyInfoInDB(slot primitives.Slot) (primitives.Slot,
 		}
 	}
 
-	earliestAvailableSlot, custodyGroupCount, err := s.cfg.BeaconDB.UpdateCustodyInfo(s.ctx, slot, custodyGroupCount)
+	earliestAvailableSlot, actualCustodyGroupCount, err := s.cfg.BeaconDB.UpdateCustodyInfo(s.ctx, slot, targetCustodyGroupCount)
 	if err != nil {
 		return 0, 0, errors.Wrap(err, "update custody info")
 	}
 
-	return earliestAvailableSlot, custodyGroupCount, nil
+	if !wasSupernode && isSupernode && actualCustodyGroupCount < cfg.NumberOfCustodyGroups {
+		log.WithFields(logrus.Fields{
+			"current":            actualCustodyGroupCount,
+			"target":             cfg.NumberOfCustodyGroups,
+			"numberOfSubnets":    cfg.NumberOfCustodyGroups,
+			"dataColumnsPerSlot": cfg.NumberOfCustodyGroups,
+		}).Info("Supernode mode enabled. Will custody all data columns going forward.")
+	}
+
+	if wasSupernode && !isSupernode {
+		log.WithField("flag", flags.SubscribeAllDataSubnets.Name).Warn(
+			"Because the flag was previously used, the node will still subscribe to all data subnets.",
+		)
```

Comment:
```suggestion
		log.Warningf("Because the `--%f` flag was previously used, the node will still subscribe to all data subnets.", flags.SubscribeAllDataSubnets.Name)
```